GVFS/GVFS.Common/Database/SqliteDatabase.cs

```diff
@@ -0,0 +1,65 @@
+﻿using GVFS.Common.FileSystem;
+using Microsoft.Data.Sqlite;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace GVFS.Common.Database
+{
+    public static class SqliteDatabase
+    {
+        public static bool HasIssue(string databasePath, PhysicalFileSystem filesystem, out string issue)
+        {
+            issue = null;
+
+            if (filesystem.FileExists(databasePath))
+            {
+                List<string> integrityCheckResults = new List<string>();
+
+                try
+                {
+                    string sqliteConnectionString = CreateConnectionString(databasePath);
+                    using (SqliteConnection integrityConnection = new SqliteConnection(sqliteConnectionString))
+                    {
+                        integrityConnection.Open();
+
+                        using (SqliteCommand pragmaCommand = integrityConnection.CreateCommand())
+                        {
+                            pragmaCommand.CommandText = $"PRAGMA integrity_check;";
+                            using (SqliteDataReader reader = pragmaCommand.ExecuteReader())
+                            {
+                                while (reader.Read())
+                                {
+                                    integrityCheckResults.Add(reader.GetString(0));
+                                }
+                            }
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    issue = $"Exception while trying to access {databasePath}: {e.Message}";
+                    return true;
+                }
+
+                // If pragma integrity_check finds no errors, a single row with the value 'ok' is returned
+                // http://www.sqlite.org/pragma.html#pragma_integrity_check
+                if (integrityCheckResults.Count != 1 || integrityCheckResults[0] != "ok")
+                {
+                    issue = string.Join(",", integrityCheckResults);
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public static string CreateConnectionString(string databasePath)
+        {
+            // Share-Cache mode allows multiple connections from the same process to share the same data cache
+            // http://www.sqlite.org/sharedcache.html
+            return $"data source={databasePath};Cache=Shared";

```

Comment:
Have you tried running the new database without shared-cache mode?  The data-access pattern is different (than BlobSizes) and I'm not sure we can assume that shared-cache will be better.

Reply:
I did a perf run without the shared cache and everything was slower, reading files, enumerating, as well as git checkouts so keeping this on is definitely better.